### ISSUE TITLE
Redis memory leak.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#2119]
To reproduce the problem, do the following

```python
import ray
import numpy as np

ray.init()

@ray.remote
def f(x):
    return 1

x = np.zeros(1000)

for _ in range(100000): f.remote(x)
```

Now the non-primary Redis shard should be using a decent amount of memory. Check how much it is using with `ps aux | grep "redis-server "`.

Attach a Redis client to it with

```
redis-cli -p <PORT>
```

You can get the port from the output of `ray.init()` or from `ps aux | grep "redis-server "`.

In the Redis client, do `FLUSHALL` to flush all of the keys.

Check how much memory the Redis server is using. It's still using the same amount.

**Comment by robertnishihara** [comment]
cc @twanthony

**Comment by robertnishihara** [comment]
Fixed by #1797.
